web-client/public/js/update-app.js

```diff
@@ -134,24 +137,45 @@ const synchronizeHideAllWeights = () => {
     $(WEIGHTS_HIDE_CLASS).addClass("selected");
 };
 
-const enableColorOptimal = function () {
+const enableColorOptimal = () => {
     $(BLACK_EDGES).removeClass(ACTIVE_COLOR_OPTION);
     $(BLACK_EDGES + ">span").removeClass("glyphicon-ok invisible");
     $(COLOR_EDGES).addClass(ACTIVE_COLOR_OPTION);
     $(COLOR_EDGES + ">span").addClass("glyphicon-ok");
 };
 
-const disableColorOptimal = function () {
+const disableColorOptimal = () => {
     $(COLOR_EDGES).removeClass(ACTIVE_COLOR_OPTION);
     $(COLOR_EDGES + ">span").removeClass("glyphicon-ok invisible");
     $(BLACK_EDGES).addClass(ACTIVE_COLOR_OPTION);
     $(BLACK_EDGES + ">span").addClass("glyphicon-ok");
 };
 
-export const updateSliderDisplayedValue = (slider, element) => {
+const updateSliderDisplayedValue = (slider, element) => {
```

Comment:
my problem here is that most of grnState's properties are T/F or a set of defined values, with certain defined actions for update-app to take. However, I don't understand how to trigger an action based on a change to the grnState value for each slider. The method that I could think of would be to create a true/false property in grnstate, however this would likely circumvent the MVC model. I would set the property as equal to true if there was a change in setup-handlers, however, I would have to set it back to false after I ran `updateSliderDisplayedValue` which would be contained in setup-handlers

Reply:
At face value, `updateSliderDisplayedValue` takes the value stored in `element` and transfers that, possibly with an appended `0`, to the slider. So, the model in question is the said value. What you are seeing here is a view-to-view copy of data (`element` going to `slider`).

To MVC this, the _value itself_ is what goes in `grnState`. And then, when that value changes, that `value` should go to _both_ `element` and `slider`.

So the flow is: `grnState` holds the master copy of the value (whatever it is). When it updates the app, this value is placed in both `element` and `slider` (whatever those are). If `element` changes this value (which is what seems to be implied because the value is being copied from `element`), that updates `grnState` to the new value. `updateApp` then sets that new value to (again) both `element` and `slider`.

That might be easier drawn with a picture but see if that description clears that up.